### PR TITLE
Add calculations to find if RA/Dec are within limits

### PR DESCRIPTION
**Pull request by cquiroz**

### CODE REVIEW

File: modules/core/shared/src/main/scala/lucuma/core/model/SiteCoordinatesLimits.scala [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  raStart:  RightAscension,
+  raEnd:    RightAscension,
+  decStart: Declination,
+  decEnd:   Declination
+) derives Eq {
+  def inRaLimits(ra: RightAscension): Boolean =
+    if (raStart < raEnd)
+      raStart <= ra && ra <= raEnd
+    else
+      raStart <= ra || (ra.toHourAngle.toDoubleDegrees >= 0 && ra <= raEnd)
```
Comment (swalker2m) [inline modules/core/shared/src/main/scala/lucuma/core/model/SiteCoordinatesLimits.scala:23]: Is the `0` check necessary?  Seems like an RA should always be `[0, 24)`.

Comment (cquiroz) [reply modules/core/shared/src/main/scala/lucuma/core/model/SiteCoordinatesLimits.scala:23]: I took this from the phase1 model code but I was wondering the same as I typed.
I can take it away

Comment (swalker2m) [reply modules/core/shared/src/main/scala/lucuma/core/model/SiteCoordinatesLimits.scala:23]: It seems like we should include an assertion in the `HourAngle` "constructor", but all ways to make an `HourAngle` appear to produce a result that is in range.

**Review by swalker2m — APPROVED** [review]
Looks great. :+1:

**Pull request merged**